packages/resource-deployment/scripts/process-utilities.sh

```diff
@@ -12,20 +12,69 @@ function waitForProcesses() {
     for pid in "${!list}"; do
         echo "Waiting for process with pid $pid"
         wait $pid
-        echo "Process with pid $pid exited"
+        local pExitCode=$?
+        echo "Process with pid $pid exited with exit code $pExitCode"
+
+        if [[ $pExitCode != 0 ]]; then
+            echo "Process - $pid failed with exit code $pExitCode. Killing current proccess."
+            exit $pExitCode
+        fi
+
     done
 }
 
-function runInParallel() {
-    local processPaths=$1
+function runCommandsWithoutSecretsInParallel {
+    local commands=$1
     local -a parallelizableProcesses
 
-    list="$processPaths[@]"
-    for processPath in "${!list}"; do
-        . "${processPath}" &
-        echo "Created process with pid $! for process path - $processPath"
+    local list="$commands[@]"
+    for command in "${!list}"; do
+        eval "$command" &
+        echo "Created process with pid $! for command - $command"
         parallelizableProcesses+=("$!")
     done
 
     waitForProcesses parallelizableProcesses
+} 
+
+function sendSignalToProcessIfExists {
+    local currentPid=$1	
+    local signal=$2
+
+    if [[ -z $currentPid ]]; then	
+        return	
+    fi	
+
+    if kill -0 $currentPid > /dev/null 2>&1; then	
```

Comment:
when we kill arm template related process azure will keep deployment running and we may need to go to azure to delete a deployment; otherwise it can take much longer than normal deployment flow, example - api management deployment cannot be re-run in this scenario

Reply:
yep. aware of this issue. May be i can print that in the console log.

Reply:
it is not supported in az cli https://github.com/Azure/azure-cli/issues/12252

Reply:
seems not implemented yet

Reply:
added WARN log in onExit function in install_parallel.

Reply:
We need to postpone the change with kill implementation till we have reliable way to terminate pending deployment on Azure